src/main/java/org/folio/consortia/service/impl/PublicationServiceImpl.java

```diff
@@ -162,53 +179,51 @@ private PublicationTenantRequestEntity savePublicationTenantRequest(PublicationT
     }
   }
 
-  CompletableFuture<ResponseEntity<String>> executeAsyncHttpRequest(PublicationRequest publicationRequest, String tenantId, FolioExecutionContext centralTenantContext) {
-    return CompletableFuture.supplyAsync(() -> {
-        try (var context = new FolioExecutionContextSetter(prepareContextForTenant(tenantId, folioModuleMetadata, centralTenantContext))) {
-          var response = httpRequestService.performRequest(publicationRequest.getUrl(), HttpMethod.POST, publicationRequest.getPayload());
-          if (response.getStatusCode().is2xxSuccessful()) {
-            log.info("executeAsyncTask:: successfully called {} on tenant {}", publicationRequest.getUrl(), tenantId);
-          } else {
-            var errMessage = response.getBody() != null ? response.getBody() : "Generic Error";
-            log.error("executeAsyncTask:: error making {} '{}' request on tenant {}", publicationRequest.getMethod(), publicationRequest.getUrl(), tenantId, new HttpException(errMessage));
-            throw new HttpClientErrorException(response.getStatusCode(), response.getBody());
-          }
-          return response;
-        } catch (HttpClientErrorException e) {
-          log.error("executeAsyncTask:: error making {} '{}' request on tenant {}", publicationRequest.getMethod(), publicationRequest.getUrl(), tenantId, new HttpException(e.getMessage()));
-          throw new HttpClientErrorException(e.getStatusCode(), e.getMessage());
-        }
-      });
+  ResponseEntity<String> executeHttpRequest(PublicationRequest publicationRequest, String tenantId, FolioExecutionContext centralTenantContext) {
+    try (var context = new FolioExecutionContextSetter(prepareContextForTenant(tenantId, folioModuleMetadata, centralTenantContext))) {
+      var response = httpRequestService.performRequest(publicationRequest.getUrl(), HttpMethod.POST, publicationRequest.getPayload());
+      if (response.getStatusCode().is2xxSuccessful()) {
+        log.info("executeAsyncTask:: successfully called {} on tenant {}", publicationRequest.getUrl(), tenantId);
+      } else {
+        var errMessage = response.getBody() != null ? response.getBody() : "Generic Error";
+        log.error("executeAsyncTask:: error making {} '{}' request on tenant {}", publicationRequest.getMethod(), publicationRequest.getUrl(), tenantId, new HttpException(errMessage));
```

Comment:
1) method name in logs should be updated
2) maybe log same exception as thrown?